src/ctelnet.cpp

```diff
@@ -1184,10 +1184,16 @@ void cTelnet::setGMCPVariables(const QString& msg)
 {
     QString packageMessage;
     QString data;
-    packageMessage = msg.section(QChar::Space, 0, 0);
-    data = msg.section(QChar::Space, 1);
 
-    if (data.isEmpty()) {
+    int firstNewline = msg.indexOf(QChar::LineFeed);
+    int firstSpace = msg.indexOf(QChar::Space);
+
+    // if we see a space before a newline, or no newlines at all,
+    // then that's the separator for message and data
+    if (Q_LIKELY((firstSpace != -1 && firstSpace < firstNewline) || firstNewline == -1)) {
```

Comment:
I don't think `firstSpace < firstNewline` is correct if there are only spaces but no newlines as `firstNewLine` would be `-1` and thus smaller than any spaces.

Maybe using `msg.indexOf(QRegExp("\\s+")` would be better? You can also use that regexp for the `section` below and get rid of the `if` overall

Reply:
> I don't think firstSpace < firstNewline is correct if there are only spaces but no newlines as firstNewLine would be -1 and thus smaller than any spaces.

If there are only spaces then the separator must be a space, no?

Reply:
Oh I see... I someow forgot to apply the `||` part... That shows that the if clause is not very explicit (to me), but I have no better idea either...

Reply:
If you introduce a regex, you have to scan the string (until the first separator) only once (for section) instead of 3 times (once for whitespaces, once for spaces, then for section).

Reply:
Yes... okay. I'll change it.